src/common/UserProvider/UserProvider.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import {
+  User,
+  UserCurrentStatus,
+  UserProviderState,
+  GatekeeperRequestError,
+} from "./interfaces";
+import PropTypes from "prop-types";
+import useSWR from "swr";
+import { fetcher } from "../../libs";
+
+export const UserProviderContext = React.createContext({} as UserProviderState);
+
+/**
+ * Provider that provides user state to any child react element
+ */
+export const UserProvider: React.FC<React.HTMLProps<HTMLDivElement>> = ({
+  children,
+}) => {
+  const response = useSWR<User | GatekeeperRequestError>("/auth/user", fetcher);
```

Comment:
This refreshes every 3 seconds by default right? Might be nice having a comment that says that so we don't have to check SWR's documentation

Reply:
Looking into the docs, it actually wont refresh on an interval if you don't specify it to, so by default in only refreshes when the user puts focus back on the website (by tabbing back in, etc). I will add this comment to the code however